pegasus/sites.v3/code.org/public/js/volunteer-local.js

```diff
@@ -0,0 +1,325 @@
+      var contact_link = '<a id="contact-trigger-' + index + '" class="contact-trigger" onclick="return contactVolunteer()">Contact</a>';
+
+      var location = {
+        lat: lat,
+        lon: lon,
+        title: title,
+        id: id,
+        contact_title: contact_title,
+        html: html + contact_link,
+        zoom: 10
+      };
+
+      locations.push(location);
+    }
+  }
+
+  return locations;
+}
+
+function resetFacets() {
+  $.each(selectize, function(key, select) {
+    select.selectize.clear();
+    select.selectize.refreshOptions(false);
+  });
+}
+
+function updateFacets(results) {
+  var facet_fields = results.facet_counts.facet_fields;
+}
+
+function displayNoResults() {
+  $('#controls').html('<p>No results were found.</p>');
+
+  // Hide the facets by default.
+  $('#volunteer-search-facets').hide();
+
+  // If a facet has a value, show the facets.
+  var form_data = $('#volunteer-search-form').serializeArray();
+  $.each(form_data, function(key, field) {
+    if (field.name != 'location' && field.value) {
+      $('#volunteer-search-facets').show();
+    }
+  });
+}
+
+function displayQueryError() {
+  $('#volunteer-search-facets').hide();
+  $('#volunteer-search-results').hide();
+  $('#volunteer-search-error').html('<p>An error occurred. Please try your search again.</p>').show();
+}
+
+function loadMap(locations) {
+  var coordinates = gmap_loc.split(',');
+  var lat = coordinates[0];
+  var lng = coordinates[1];
+
+  // Reset the map.
+  $('#gmap').html('');
+  gmap = new Maplace;
+
+  var mapOptions = {
+    mapOptions: {
+      setCenter: [lat, lng],
+      zoom: 12
+    },
+    controls_type: 'list',
+    controls_on_map: false
+  };
+
+  if (locations.length > 0) {
+    mapOptions.forceGenerateControls = true;
+    mapOptions.locations = locations;
+    mapOptions.afterOpenInfowindow = function(index, location, marker) {
+      setContactTrigger(index, location, marker);
+    };
+  }
+
+  gmap.Load(mapOptions);
+}
+
+function compileHTML(index, location) {
+  var lines = [];
+  var line;
+
+  // Compile HTML.
+  var html = '<h3>' + location.name_s + '</h3>';
+
+  if (location.company_s) {
+    line = location.company_s;
+    lines.push(line);
+  }
+
+  if (location.experience_s) {
+    line = '<strong>Experience: </strong>' + i18n(location.experience_s);
+    lines.push(line);
+  }
+
+  if (location.location_flexibility_ss) {
+    $.each(location.location_flexibility_ss, function(key, field) {
+      location.location_flexibility_ss[key] = i18n('location_' + field);
+    });
+
+    line = '<strong>How I can help: </strong>' + location.location_flexibility_ss.join(', ');
+    lines.push(line);
+  }
+
+  if (location.description_s) {
+    line = '<strong>About me: </strong>' + location.description_s;
+    lines.push(line);
+  }
+
+  if (location.linkedin_s) {
+    if (!location.linkedin_s.match(/^https?:\/\//i)) {
+      location.linkedin_s = 'http://' + location.linkedin_s;
```

Comment:
Identifier 'linkedin_s' is not in camel case.<br>Too many errors. (68% scanned).